python/servo/testing_commands.py

```diff
@@ -556,6 +566,37 @@ def filter_intermittents(self, summary, log_filteredsummary, log_intermittents,
                 else:
                     intermittents += [failure]
 
+        if reporter_api:
+            if reporter_api.endswith('/'):
+                reporter_api = reporter_api[0:-1]
+            reported = set()
+            for intermittent in intermittents:
+                if intermittent['test'] in reported:
+                    continue
+                reported.add(intermittent['test'])
+
+                proc = subprocess.Popen(
+                    ["git", "log", "--merges", "--oneline", "-1"],
+                    stdout=subprocess.PIPE)
+                (last_merge, _) = proc.communicate()
+
+                # Extract the issue reference from "abcdef Auto merge of #NNN"
+                pull_request = last_merge.split(' ')[4][1:]
```

Comment:
A bunch of this looks like it could be pulled out of the `for intermittent in intermittents` loop; I presume you'd want to add an early return if there are no intermittents as well.